addin-proximity/app/scripts/main.js

```diff
@@ -451,48 +449,53 @@ geotab.addin.proximity = () => {
         sizeChanged(300);
 
         // initialize multiselect/autocomplte
-        vehicleMultiselect = new Choices(elVehicleSelect, { removeItemButton: true, duplicateItemsAllowed: false, searchResultLimit: 50, noChoicesText: 'Start typing to search for a vehicle'});
+        vehicleMultiselect = new Choices(elVehicleSelect, {removeItemButton: true, duplicateItemsAllowed: false, searchResultLimit: 1000, maxItemCount: 1000, noChoicesText: 'Start typing to search for vehicles'});
 
         // events
         elVehicleMultiSelectContainer.addEventListener('keyup', debounce(e => {
-            let manualSearch = '%%';
-            let deviceList = [];
-            manualSearch = '%' + e.target.value + '%';
 
-            api.call('Get', {
-                typeName: 'Device',
-                search: {
-                    fromDate: new Date().toISOString(),
-                    groups: state.getGroupFilter(),
-                    name: manualSearch,
-                }
-            }, newDevices => {
-                if (!newDevices || newDevices.length < 1) {
-                    return;
-                };
-
-                for(var i=0;i<newDevices.length;i++){
-                    if(selected.includes(newDevices[i].id)){
-                        continue;
-                    }
-                    else{
-                        deviceList.push(newDevices[i])
-                    }
-                }  
+            vehicleMultiselect.clearChoices();
+            let manualSearch = '%' + e.target.value + '%';
+            let deviceList = [];
+            console.log('manualSearch: ', manualSearch);
+            
+            if(manualSearch!='%%'){
```

Comment:
Can use more strict `manualSearch !== '%%'`

Reply:
Do we care if it's an empty search if we are applying resultsLimit?

Reply:
Yes the conditional for the empty search matters. If there is an empty search, the choices will reset and the options will say 'Start typing to search for vehicles.' Without it, the last search vehicles are displayed and it's confusing to know what is being displayed. By resetting we are avoiding the end user from getting confused.